jbpm-benchmarks/kieserver-performance-tests/src/main/java/org/jbpm/test/performance/taskassigning/TaskAssigning.java

```diff
@@ -0,0 +1,121 @@
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.jbpm.test.performance.kieserver.KieServerClient;
+import org.jbpm.test.performance.kieserver.KieServerTestConfig;
+import org.kie.server.api.model.definition.QueryDefinition;
+import org.kie.server.api.model.instance.ProcessInstance;
+import org.kie.server.client.ProcessServicesClient;
+import org.kie.server.client.QueryServicesClient;
+import org.kie.server.client.UserTaskServicesClient;
+
+/**
+ * A Common ground for task assigning load test scenarios. Before running any of these scenarios, following
+ * preconditions have to be met:
+ * <li>
+ *      Two KIE servers are configured; the first one hosts a task planning extension and the second one task assignment
+ *      and jBPM extensions. However, the scenario communicates only with the jBPM extension via KIE server remote API.
+ * </li>
+ * <li>
+ *     KIE server task planning extension is configured to optimize for a planning window that can contain all tasks
+ *     created in this scenario.
+ * </li>
+ */
+public abstract class TaskAssigning {
+    protected static final String CONTAINER_ID = "kieserver-assets";
+    protected static final String PROCESS_ID = "test-jbpm-assignment.testTaskAssignment";
+
+    private static final String DATA_SOURCE_JNDI = KieServerTestConfig.getInstance().getDataSourceJndi();
+
+    private final KieServerClient client;
+    private final ProcessServicesClient processClient;
+    private final UserTaskServicesClient taskClient;
+    private final QueryServicesClient queryClient;
+
+    public TaskAssigning() {
+        // Must be set before the KIE Server client is created.
+        System.setProperty("org.kie.server.bypass.auth.user", Boolean.toString(true));
+        client = new KieServerClient();
+        processClient = client.getProcessClient();
+        taskClient = client.getTaskClient();
+        queryClient = client.getQueryClient();
+    }
+
+    protected void registerQuery(String queryName, String queryCode) {
+        QueryDefinition query = new QueryDefinition();
+        query.setName(queryName);
+        query.setSource(DATA_SOURCE_JNDI);
+        query.setExpression(queryCode);
+        query.setTarget("CUSTOM");
+        queryClient.replaceQuery(query);
+    }
+
+    protected void abortAllProcesses() {
+        List<ProcessInstance> processInstanceList = processClient.findProcessInstances(CONTAINER_ID, 0, Integer.MAX_VALUE);
+        List<Long> processInstanceIdList = processInstanceList.stream()
+                .map(processInstance -> processInstance.getId())
+                .collect(Collectors.toList());
+        processClient.abortProcessInstances(CONTAINER_ID, processInstanceIdList);
+    }
+
+    protected Collection<CompletableFuture<Void>> startProcessesAsync(int processCount, int numberOfThreads) {
```

Comment:
Do you want to keep that method now you don't use async start of processes?

Reply:
No, if it's not used, no reason for it to stay.